assets/src/components/map.tsx

```diff
@@ -0,0 +1,111 @@
+import Leaflet, { Map as LeafletMap, Marker } from "leaflet"
+import "leaflet-rotatedmarker"
+import React, { MutableRefObject, ReactElement, useEffect, useRef } from "react"
+import { renderToStaticMarkup } from "react-dom/server"
+import { Label, Orientation, Size } from "./vehicleIcon"
+
+interface Props {
+  bearing: number
+  label: string
+  latitude: number
+  longitude: number
+}
+
+const iconAnchor: [number, number] = [9, 3]
+
+export const updateMap = (
+  { bearing, label: labelText, latitude, longitude }: Props,
+  existingMap: LeafletMap | null,
+  existingVehicleMarker: Marker | null,
+  existingVehicleLabel: Marker | null
+): {
+  map: LeafletMap | null
+  vehicleMarker: Marker | null
+  vehicleLabel: Marker | null
+} => {
+  if (!latitude || !longitude) {
+    return {
+      map: existingMap,
+      vehicleMarker: existingVehicleMarker,
+      vehicleLabel: existingVehicleLabel,
+    }
+  }
+
+  const map =
+    existingMap ||
+    Leaflet.map("map", {
+      layers: [
+        Leaflet.tileLayer(
+          `https://mbta-map-tiles-dev.s3.amazonaws.com/osm_tiles/{z}/{x}/{y}.png`,
+          {
+            attribution:
+              '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+          }
+        ),
+      ],
+      zoom: 16,
+    })
+
+  const vehicleMarker =
+    existingVehicleMarker ||
+    Leaflet.marker([latitude, longitude], {
+      rotationOrigin: `${iconAnchor[0]}px ${iconAnchor[1]}px`,
```

Comment:
I see this requires another dependency to do. Would it be possible to do this by rotating the SVG instead of the marker? Could that save us from adding this dependency?

Reply:
I had trouble rotating the marker initially using transform, but the code was pretty different at that point. i'll try it again.